docs/magento-layout-loader.md

```diff
@@ -0,0 +1,288 @@
+| `operation`       |                         One of the supported types of operations                         |
+| `targetContainer` |                    The `data-mid` value of the `Container` to target                     |
+| `targetChild`     |        The `id` value of the `ContainerChild` to target within `targetContainer`         |
+| `componentPath`   | An absolute path pointing to a file containing a React component as the `default` export |
+
+### Operations
+
+#### removeContainer
+
+##### Configuration
+
+```js
+{
+    "operation": "removeContainer",
+    "targetContainer": "any.container.id"
+}
+```
+
+<details>
+<summary>Example</summary>
+
+##### Input
+
+```js
+import React from 'react';
+
+function render() {
+    return (
+        <div className="wrapper">
+            The div below will be removed
+            <div data-mid="any.container.id" />
+        </div>
+    );
+}
+```
+
+##### Output
+
+```js
+import React from 'react';
+
+function render() {
+    return <div className="wrapper">The div below will be removed</div>;
+}
+```
+
+</details>
+
+#### removeChild
+
+##### Configuration
+
+```js
+{
+    "operation": "removeChild",
+    "targetContainer": "any.container.id",
+    "targetChild": "container.child.id"
+}
+```
+
+<details>
+<summary>Example</summary>
+
+##### Input
+
+```js
+import React from 'react';
+import { ContainerChild } from '@magento/peregrine';
+
+function render() {
+    return (
+        <div className="wrapper" data-mid="any.container.id">
+            The container below will be removed
+            <ContainerChild
+                id="container.child.id"
+                render={() => <div>This content will be removed</div>}
+            />
+        </div>
+    );
+}
+```
+
+##### Output
+
+```js
+import React from 'react';
+
+function render() {
+    return (
+        <div className="wrapper" data-mid="any.container.id">
+            The container below will be removed
+        </div>
+    );
+}
+```
+
+</details>
+
+#### insertBefore
+
+##### Configuration
+
+```js
+{
+    "operation": "insertBefore",
+    "targetContainer": "any.container.id",
+    "targetChild": "container.child.id",
+    "componentPath": "/Absolute/path/to/a/component.js"
+}
+```
+
+<details>
+<summary>Example</summary>
+
+##### Input
+
+```js
+import React from 'react';
+import { ContainerChild } from '@magento/peregrine';
+
+function render() {
+    return (
+        <div className="wrapper" data-mid="any.container.id">
+            <ContainerChild
+                id="container.child.id"
+                render={() => <div>Some Content</div>}
+            />
+        </div>
+    );
+}
+```
+
+##### Output
+
+```js
+import React from 'react';
+import { ContainerChild } from '@magento/peregrine';
+import _Extension from '/Absolute/path/to/a/component.js';
+
+function render() {
+    return (
+        <div className="wrapper" data-mid="any.container.id">
+            <_Extension />
```

Comment:
I'm wondering if a magento module will be able to add an Extension and another one will be able to remove it ? This seems impossible to me right now since the added extensions do not have ids.
It would also mean that if a module adds two Extensions and another one wants to add another one in between, it might be hard (`<ExtModule1a /><ExtModule2 /><ExtModule1b /><ContainerChild {...} />`).
SoI wonder if the added Extensions shouldnt be contained within several `ContainerChild`.

However I might be a bit too early here. If so, do not consider my comment 🙂

Reply:
Thanks for the feedback. This is already being tracked in #16